fedimint-client/src/transaction/sm.rs

```diff
@@ -77,6 +78,14 @@ impl State for TxSubmissionStatesSM {
         global_context: &DynGlobalClientContext,
     ) -> Vec<StateTransition<Self>> {
         let operation_id = self.operation_id;
+        // There is no point awaiting tx until it was submitted, so
+        // `trigger_created_rejected` which does the submitting will use this
+        // channel to let the `trigger_created_accepted` which does the awaiting
+        // know when it did the submission.
+        //
+        // Submitting tx does not guarantee that it will get into consensus, so the
+        // submitting need to continue.
+        let (tx_submitted_sender, tx_submitted_receiver) = watch::channel(false);
```

Comment:
It never occurred to me that you could connect two trigger futures like that lol, slightly cursed :see_no_evil: 

Alternatively we could just make it one trigger that passes the result to the state transition fn.

Reply:
@elsirion TBH it seemed perfectly normal and natural to me. :D